diff --git a/configure.ac b/configure.ac
index 2cf9a19..5de3dc2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -29,12 +29,6 @@ AC_CONFIG_SRCDIR([Makefile.am])
 AC_CONFIG_MACRO_DIR([m4])
 AC_CONFIG_AUX_DIR([build-aux])
 
-# Require xorg-macros minimum of 1.12 for XORG_WITH_XSLTPROC
-m4_ifndef([XORG_MACROS_VERSION],
-          [m4_fatal([must install xorg-macros 1.12 or later before running autoconf/autogen])])
-XORG_MACROS_VERSION(1.12)
-XORG_WITH_XSLTPROC
-XORG_MANPAGE_SECTIONS
 
 AM_INIT_AUTOMAKE([1.10 foreign dist-bzip2])
 
